### ISSUE TITLE
Characters: Spend XP

### ISSUE THREAD

**Issue by Rillaith** [Rillaith/gcms#43]
Business Rules per class - offer only available upgrades and indicate cost for each in form

Future option: Allows custom classing for individuals where necessary due to rituals

**Comment by Rillaith** [comment]
Custom Classing: Shouldn't need to; there's currently only one character in play that would require this, and this can be done by GOD.

**Comment by Rillaith** [comment]
Built front end layout for Adept class.

TODO: 
- Javascript button selection colour changes (default to success class tag when selected, and recverse when unselected)
- Hidden form fields
- Javascript to update said form fields
- Regimental Skill cost/location variance
- toggle xp spend available only after an event attended (needs thinking about)

**Comment by Rillaith** [comment]
Javascript done for button click toggle colour, self-identify tier, self-identify type, toggle unbought earlier skills.

To do: show XP spend of selected buttons; cap spending; disable anything too expensive; regimental skills; form fields

**Comment by Rillaith** [comment]
Todo: Looks like the button javascript will rationalise itself to generic for all classes ; if so, split into separate file & include.

**Comment by Rillaith** [comment]
Done: form fields

Hit a blocker: need available XP spend before capping; sidetracked to prep SQL for database calculation of xp spent and vetpicks used. Now that's done, need to get "available to spend" data from user record.

**Comment by Rillaith** [comment]
Front end complete. Regimental Skill move to primary / secondary capability added. Validation of skill cost within available points done. Split script off into separate file. Pulls in available to spend from user view (conveniently - character model). 

Chosen to not disable unafforable skills as being able to see what it would cost is actually useful.

Remaining - to do: Save skill spend to database.
Remaining - migrate to separate issue: Toggle xp spend available only after an event attended (needs thinking about).

**Comment by Rillaith** [comment]
XP Spend linked to events: #67

**Comment by Rillaith** [comment]
Server side maths / validation now working, and skill spend saves to database and correctly updates all xp pools.

**Comment by Rillaith** [comment]
All classes now have their layout for progression XP spend
